### PR TITLE
Automate benchmarks

### PR DESCRIPTION
**Pull request by milkysoft**
Automate benchmarks

### CODE REVIEW

File: run_benchmarks.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+TMPDIR="${PWD}/bundles"
```
Comment (g4s8) [inline run_benchmarks.sh:1]: @milkysoft please use bash shebang https://stackoverflow.com/a/10383546

Comment (g4s8) [inline run_benchmarks.sh:1]: @milkysoft it'd be a good idea to exit the script on error by adding `set -e` at the beginning: https://stackoverflow.com/a/19622569

File: run_benchmarks.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+if [ -d "$TMPDIR" ]; then
+  # Take action if $DIR exists. #
+  echo "removing ${TMPDIR} before running benchmarks"
+  rm -rf ${TMPDIR}
+fi
+mkdir ${TMPDIR}
```
Comment (g4s8) [inline run_benchmarks.sh:7]: @milkysoft PWD is a repository location, for temp directories you can use `TMPDIR=$(mktemp --directory)`

File: run_benchmarks.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+if [ -d "$DIR" ]; then
+  # Take action if $DIR exists. #
+  echo "remove ${DIR} before running benchmarks"
+  exit 0
+fi
+echo "Currently TMP: ${TMPDIR}"
+#cp /mnt/disk2/projects/rpmtests/bundle1000.tar.gz ${TMPDIR}
+#cp /mnt/disk2/projects/rpmtests/bundle100.tar.gz ${TMPDIR}
+
+git clone https://github.com/artipie/rpm-adapter.git
```
Comment (g4s8) [inline run_benchmarks.sh:19]: @milkysoft this script is a part of repository, so I think we can assume the repository was already cloned here

File: run_benchmarks.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
+TMPDIR="${PWD}/bundles"
+if [ -d "$TMPDIR" ]; then
+  # Take action if $DIR exists. #
+  echo "removing ${TMPDIR} before running benchmarks"
+  rm -rf ${TMPDIR}
+fi
+mkdir ${TMPDIR}
+
+DIR="rpm-adapter"
```
Comment (g4s8) [inline run_benchmarks.sh:9]: @milkysoft what is the use case of this variable? It's not used later

File: run_benchmarks.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#cp /mnt/disk2/projects/rpmtests/bundle100.tar.gz ${TMPDIR}
+
+git clone https://github.com/artipie/rpm-adapter.git
+cd rpm-adapter
+echo ${PWD}
+mvn package -Pbench
+wget https://artipie.s3.amazonaws.com/rpm-test/bundle100.tar.gz
+tar -xvzf bundle100.tar.gz -C ${TMPDIR}
+wget https://artipie.s3.amazonaws.com/rpm-test/bundle1000.tar.gz
+tar -xvzf bundle1000.tar.gz -C ${TMPDIR}
```
Comment (g4s8) [inline run_benchmarks.sh:26]: @milkysoft we need only one bundle, not both. It could be configurable parameter with `bundle1000` default value

**Review by g4s8 — CHANGES_REQUESTED** [review]
@milkysoft please check my comment. Also, the reason of this script was a benchmarks reporting automation but I don't see how this script automate the **reporting** of the benchmarks, we need some logs parsing or maybe benchmark format configuration to report the result to the user

**Comment by milkysoft** [comment]
> @milkysoft please check my comment. Also, the reason of this script was a benchmarks reporting automation but I don't see how this script automate the **reporting** of the benchmarks, we need some logs parsing or maybe benchmark format configuration to report the result to the user

@g4s8, last two lines like:
```
Benchmark     Mode  Cnt  Score   Error  Units
RpmBench.run  avgt  250  5.822 ± 0.169  ms/op
```
are reported to the user with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b)

File: run_benchmarks.sh (hunk 1 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , fixed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b)

File: run_benchmarks.sh (hunk 2 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , fixed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b)

File: run_benchmarks.sh (hunk 3 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , fixed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b)

File: run_benchmarks.sh (hunk 4 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , fixed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b)

File: run_benchmarks.sh (hunk 5 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , `-b` optional parameter added with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/7607ca7c554416dcdf9bcb7723af6fc9e1f3c09b), which could be manually set to `bundle100` or bundle1000`.

File: run_benchmarks.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+TMPDIR=$(mktemp --directory)
+
+#echo "Currently TMP: ${TMPDIR}"
+#echo "Bundle: ${BUNDLE}"
+#cp /mnt/disk2/projects/rpmtests/bundle1000.tar.gz ${TMPDIR}
+#cp /mnt/disk2/projects/rpmtests/bundle100.tar.gz ${TMPDIR}
+
+#git clone https://github.com/artipie/rpm-adapter.git
+#cd rpm-adapter
```
Comment (g4s8) [inline run_benchmarks.sh:20]: @milkysoft please remove unused code

File: run_benchmarks.sh [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+mvn package -Pbench
+wget https://artipie.s3.amazonaws.com/rpm-test/${BUNDLE}.tar.gz
+tar -xvzf ${BUNDLE}.tar.gz -C ${TMPDIR}
+mvn dependency:copy-dependencies
+num=$(echo ${BUNDLE} | cut -c7-11)
+echo "num: ${num}"
+echo ${TMPDIR}/bundle/${num}
```
Comment (g4s8) [inline run_benchmarks.sh:27]: @milkysoft what is the meaning of these logs?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@milkysoft please see my comments

File: run_benchmarks.sh (hunk 7 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , removed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/8402f586a96209f7ac4ce19ab33711ee40c8bb28)

File: run_benchmarks.sh (hunk 6 above)
Comment (milkysoft) [reply run_benchmarks.sh]: @g4s8 , removed with latest [commit](https://github.com/artipie/rpm-adapter/pull/386/commits/8402f586a96209f7ac4ce19ab33711ee40c8bb28)

**Review by g4s8 — APPROVED** [review]
Thanks

**Pull request merged**